nft_ingester/src/bin/synchronizer/main.rs

```diff
@@ -139,64 +140,33 @@ pub async fn main() -> Result<(), IngesterError> {
                     }
                 }
             }
+            while shutdown_rx.is_empty() {
+                let result = synchronizer
+                    .synchronize_asset_indexes(asset_type, &shutdown_rx, config.dump_sync_threshold)
+                    .await;
+    
+                match result {
+                    Ok(_) => {
+                        tracing::info!("{:?} Synchronization finished successfully", asset_type)
+                    }
+                    Err(e) => tracing::error!("{:?} Synchronization failed: {:?}", asset_type, e),
+                }
+    
+                tokio::time::sleep(tokio::time::Duration::from_secs(
+                    config.timeout_between_syncs_sec,
```

Comment:
shutdown will have to wait if it receives the signal.

Reply:
Change to tokio::select over sleep and cancelation token. Please let me know what you think of this approach